core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -630,5 +632,23 @@ class CityInfo {
         return !isOriginalCapital && !isCapital() && !isHolyCity()
     }
 
+
+    fun getNeighbouringCivs(): List<String> {
+        val tilesList: HashSet<TileInfo> = getTiles().toHashSet()
+        val cityPositionList: ArrayList<TileInfo> = arrayListOf()
+
+        for (tiles in tilesList)
+            for (tile in tiles.neighbors)
+                if (!tilesList.contains(tile))
+                    cityPositionList.add(tile)
+
+        val NeighbouringCivs = cityPositionList.asSequence()
```

Comment:
.map{it.getOwner()?.civName}.filterNotNull().toSet()

Reply:
Isn't `.mapNotNull{it.getOwner()?.civName}` better?

Reply:
fixed